app/assets/javascripts/controllers/middleware_datasource/middleware_datasource_controller.js

```diff
@@ -61,64 +72,95 @@ function MwAddDatasourceCtrl($scope, $rootScope, miqService, mwAddDatasourceServ
     miqService.sparkleOff();
   });
 
-  $scope.addDatasourceChooseNext = function() {
-    var dsSelection = $scope.chooseDsModel.selectedDatasource;
-    $scope.dsModel.step = 'STEP1';
-    $scope.step1DsModel.datasourceName = dsSelection.name;
-    $scope.step1DsModel.jndiName = dsSelection.jndiName;
+  $scope.$watch(angular.bind(this, function() {
+    return vm.step2DsModel.selectedJdbcDriver;
+  }), function(driverSelection) {
+    var dsSelection = mwAddDatasourceService.findDsSelectionFromDriver(driverSelection);
+    if (dsSelection) {
+      vm.step1DsModel.datasourceName = dsSelection.name;
+      vm.step1DsModel.jndiName = dsSelection.jndiName;
+      vm.step2DsModel.jdbcDriverName = dsSelection.driverName;
+      vm.step3DsModel.connectionUrl = '';
+    }
+    if (mwAddDatasourceService.isXaDriver(driverSelection)) {
+      vm.step2DsModel.xaDsClass = driverSelection.xaDsClass;
+    } else {
+      vm.step2DsModel.driverClass = driverSelection.driverClass;
+    }
+  });
+
+  vm.addDatasourceChooseNext = function() {
+    var dsSelection = vm.chooseDsModel.selectedDatasource;
+    vm.dsModel.step = 'STEP1';
+    vm.step1DsModel.datasourceName = dsSelection.name;
+    vm.step1DsModel.jndiName = dsSelection.jndiName;
   };
 
-  $scope.addDatasourceStep1Next = function() {
-    var dsSelection = $scope.chooseDsModel.selectedDatasource;
-    $scope.dsModel.step = 'STEP2';
+  vm.addDatasourceStep1Next = function() {
+    var dsSelection = vm.chooseDsModel.selectedDatasource;
+    var serverId = angular.element('#server_id').val();
```

Comment:
I'm a bit uncomfortable with abusing a hidden field for this, and wary of depending on the template in the controller.. Any chance you could move this to a provided value?

(We do stuff like `ManageIQ.angular.app.value('timeProfileFormId', '#{@timeprofile.id || "new"}');` all over the place..)